agents-for-bedrock/use-case-examples/text-2-sql-agent-cdk-enhanced/app.py

```diff
@@ -0,0 +1,225 @@
+            actions=["s3:PutObject", "s3:GetObject", "s3:ListBucket","s3:CreateBucket", "s3:GetBucketLocation"],
+            resources=[f"{schema_bucket.bucket_arn}", f"{schema_bucket.bucket_arn}/*"]
+        ))
+        action_group_function.role.add_to_policy(iam.PolicyStatement(
+            actions=[
+                "athena:StartQueryExecution", "athena:GetQueryExecution",
+                "athena:GetQueryResults", "athena:StopQueryExecution",
+                "athena:GetWorkGroup"
+            ],
+            resources=[f"arn:aws:athena:{region}:{account_id}:workgroup/primary"]
+        ))
+
+        
+        
+        data_folder = f'./Data/{data_folder_name}'
+
+        data_context = analyze_csv_files(data_folder)
+        
+        instruction_text = generate_instruction(data_context, glue_database_name)
+        # question = f"""
+        #     Please craft a comprehensive, single-paragraph instruction (up to 1200 words) for the Bedrock agent. Incorporate all the contextual details provided below to develop this instruction text. The final submission should consist solely of this detailed instruction.\n
+        #     {instruction_text}.
+        #     """ 
+            
+        
+        question = f"""
+            Craft a comprehensive and cohesive paragraph instruction for the Bedrock agent, ensuring the instruction text includes all 7 contextual details and examples provided. The instruction should be detailed, precise with a maximum length of 4000 characters. Clearly outline the agent's tasks and how it should interact with users, incorporating the provided contextual details and examples with minimal changes. Avoid any introductory phrases such as "Here is your...".
+
+            <Contextual details and examples>
+            {instruction_text}
+            """
+   
+                 
+        agent_instruction=invoke_claude_3_with_text(question)    
+        print(agent_instruction)
+        
+        api_schema = ApiSchema.from_asset("./text_to_sql_openapi_schema.json")
+        agent = Agent(
+            self,
+            "MyAgent",
+            foundation_model=foundation_model,
+            instruction=agent_instruction,
+            description="Agent for performing sql queries.",
+            should_prepare_agent=True
+          
+        )
+        # agent.add_alias(self, 'ProdAlias', 
+        #     alias_name='prod',
+        #     agent_version='1'
+        #     )   
+        action_group = AgentActionGroup(self,
+            "MyActionGroup",
+            action_group_name="QueryAthenaActionGroup",
+            description="Actions for getting the database schema and querying the Athena database for sample data or final query.",
+            action_group_executor=action_group_function,
+            action_group_state="ENABLED",
+            api_schema=api_schema)
+        
+        agent.add_action_group(action_group)
+        
+        # agent.add_action_group(
```

Comment:
I don't think this code is required

Reply:
this one?
agent.add_action_group(action_group)

Reply:
the one commented out. Either you add the action group via object or with everything written out first

Reply:
ok, just removed the commented